### PR TITLE
Support for enabling email notifications

### PR DESCRIPTION
**Pull request by dbkr**
Supports https://github.com/vector-im/vector-web/pull/1469 (this PR is safe without the vector-web PR)

https://github.com/vector-im/vector-web/issues/108

### CODE REVIEW

File: src/components/structures/UserSettings.js [hunk 1]
```diff
@@ -328,12 +328,12 @@ module.exports = React.createClass({
             );
         }
         var notification_area;
-        if (!MatrixClientPeg.get().isGuest()) {
+        if (!MatrixClientPeg.get().isGuest() && this.state.threepids !== undefined) {
```
Comment (ara4n) [inline src/components/structures/UserSettings.js]: surely we should still show the Notifications settings, even if the user has no threepids?

File: src/UserSettingsStore.js [hunk 2]
```diff
@@ -77,4 +77,33 @@ module.exports = {
...
+    addEmailPusher: function(address) {
+        return MatrixClientPeg.get().setPusher({
+            kind: 'email',
+            app_id: "m.email",
+            pushkey: address,
+            app_display_name: 'Email Notifications',
+            device_display_name: address,
+            lang: navigator.language,
+            data: {},
+            append: true,  // We always append for email pushers since the email is not a device that would 'logged out' from
```
Comment (ara4n) [inline src/UserSettingsStore.js]: the comment could do with being better english; unsure what it's trying to say

**Comment by ara4n** [comment]
Stealing this from vdh as he's on a boat.

@dbkr: LGTM, modulo comments - although I am concerned that we seem to have an asymmetry between getEmailPusher() that just returns the first pusher, whereas addEmailPusher() lets you add multiple ones.  What's the intended behaviour here for multiple email pushers?  It would be nice to comment the methods to make this clear.

File: src/components/structures/UserSettings.js (hunk 1 above)
Comment (dbkr) [inline src/components/structures/UserSettings.js]: If they have no threepids, this should be the empty list.

**Comment by dbkr** [comment]
getEmailPusher returns the first pusher for that email address - could make it return an array although it would be an extreme edge case to have to pushers for the same account and email address.

**Comment by ara4n** [comment]
@dbkr 'to have /different/ pushers for the same account and email address'?  i'm lost - perhaps some comments on getEmailPusher and addEmailPusher would help?

**Comment by dbkr** [comment]
Done - is this any clearer?

**Comment by ara4n** [comment]
lgtm

**Pull request merged**